fastplotlib/legends/legend.py

```diff
@@ -0,0 +1,344 @@
+from functools import partial
+from collections import OrderedDict
+from typing import *
+
+import numpy as np
+import pygfx
+
+from ..graphics._base import Graphic
+from ..graphics._features._base import FeatureEvent
+from ..graphics import LineGraphic, ScatterGraphic, ImageGraphic
+from ..utils import mesh_masks
+
+
+class LegendItem:
+    def __init__(
+            self,
+            label: str,
+            color: pygfx.Color,
+    ):
+        """
+
+        Parameters
+        ----------
+        label: str
+
+        color: pygfx.Color
+        """
+        self._label = label
+        self._color = color
+
+
+class LineLegendItem(LegendItem):
+    def __init__(
+            self,
+            parent,
+            graphic: LineGraphic,
+            label: str,
+            position: Tuple[int, int]
+    ):
+        """
+
+        Parameters
+        ----------
+        graphic: LineGraphic
+
+        label: str
+
+        position: [x, y]
+        """
+
+        if label is not None:
+            pass
+
+        elif graphic.name is not None:
+            pass
+
+        else:
+            raise ValueError("Must specify `label` or Graphic must have a `name` to auto-use as the label")
+
+        if np.unique(graphic.colors(), axis=0).shape[0] > 1:
+            raise ValueError("Use colorbars for multi-colored lines, not legends")
```

Comment:
Can you add code comment here explaining this better?

I assume you are checking to see if there is more than 1 color for a `Graphic`, and if so, a `Legend` is not appropriate to use?

Reply:
We made some more thoughts on this here, but for now only single colored lines in this PR https://github.com/fastplotlib/fastplotlib/issues/403#issuecomment-1961981225